app/jobs/export_job.rb

```diff
@@ -5,34 +5,56 @@ class ExportJob < ApplicationJob
   queue_as :exports
   include ImportExport
 
-  # Limits number of Patient records to be considered for a single exported file to handle maximum file size limit.
-  # Adds additional files as needed if exceeds batch size.
-  OUTER_BATCH_SIZE = ENV['EXPORT_OUTER_BATCH_SIZE']&.to_i || 10_000
-
   # Inner batch size limits number of Patient records details help in memory at once before writing to file.
-  INNER_BATCH_SIZE = ENV['EXPORT_INNER_BATCH_SIZE']&.to_i || 500
+  INNER_BATCH_SIZE = ENV.fetch('EXPORT_INNER_BATCH_SIZE', 500).to_i unless const_defined?(:INNER_BATCH_SIZE)
 
   def perform(config)
     # Get user in order to query viewable patients
     user = User.find_by(id: config[:user_id])
     return if user.nil?
 
     # Delete any existing downloads of this type
-    user.downloads.where(export_type: config[:export_type]).delete_all
+    user.downloads.where(export_type: config[:export_type]).destroy_all
 
     # Extract data
     data = config[:data]
     return if data.nil?
 
     # Construct export
+    downloads = []
+
     patients = patients_by_query(user, data.dig(:patients, :query) || {})
-    files = write_export_data_to_files(config, patients, OUTER_BATCH_SIZE, INNER_BATCH_SIZE)
-    return unless files.present?
+    files = write_export_data_to_files(config, patients.reorder(''), INNER_BATCH_SIZE)
```

Comment:
why is it necessary to call `reorder` here when it's called in `import_export` already?

Reply:
This was just the line used back when I started this - if it is not required we can remove the reorder. I started a more in depth discussion in Adam's comment.